### DISCUSSION TITLE
"Invalid Input" error on VendorOrders get_purchase_orders() depending on changedAfter value

### DISCUSSION THREAD

**Discussion by robguttman** [saleweaver/python-amazon-sp-api#434] [General]
**Describe the bug**
Calls to `VendorOrders().get_purchase_orders()` succeed or fail depending on the value of `changedAfter`.

**To Reproduce**
This call *succeeds*:
```
VendorOrders().get_purchase_orders(changedAfter="2022-03-05T15:15:48.001000+00:00", nextToken=None)
```
This call *fails*:
```
VendorOrders().get_purchase_orders(changedAfter="2022-03-04T21:00:56.001000+00:00", nextToken=None)
```
The error response is:
```
SellingApiBadRequestException: [{'code': 'InvalidInput', 'message': 'Invalid Input', 'details': ''}]
```

Notice that the only difference is the `changedAfter` value - both are in ISO-8601 date/time format and both are well within the 7 day range stated in the documentation.  These calls also *fail*:
```
VendorOrders().get_purchase_orders(changedAfter="2022-03-04T21:00:56.001000", nextToken=None)
VendorOrders().get_purchase_orders(changedAfter="2022-03-04T21:00:56.001", nextToken=None)
VendorOrders().get_purchase_orders(changedAfter="2022-03-04T21:00:56", nextToken=None)
```

**Expected behavior**
It's expected that these calls should succeed as long as the `changedAfter` date is ISO-8601 date/time format and within the 7 day range.

**Additional context**
I suspect these issues are with the Amazon REST API and not this SDK, but it's hard to know for sure.  Also, the API error response message is useless.  I also suspect this is the fault of the Amazon API and not this SDK but again, it's hard to know for sure.

**Comment by saleweaver** [comment]
Hi,

I haven't worked with the vendor api a lot, but there's nothing here that looks like this client even could cause the behavior. 

Does (datetime.utcnow() - timedelta(days=7)).isoformat() work?

**Reply by robguttman** [reply]
Michael, thanks for your speedy response.  Yes, that date does work (which BTW translates to `2022-02-26T19:21:03.066418`).

I suspect there's some bug on the Amazon API side, unless you can think of some other possible explanation?

Question: What do you recommend as the best way to submit bugs to Amazon for issues with their Selling Partner API?  The one time I submitted a bug so far using their ticketing system, they requested this info:
```
If the error persists, please create a support case including the following details:
o Application ID
o Request ID with timestamp
o API operation (please specify if it is a sandbox request)
o Error response received
```
But since I'm using this SDK, the "Request ID with timestamp" and "API operation" info isn't readily available.  Is it possible for the SDK to somehow expose this info so it can be captured and used in Amazon support tickets?  Even if it's just logging this info somewhere, that would be very helpful.

**Reply by saleweaver** [reply]
You can access request id, and all other headers in 'res.headers'. The API operation is the endpoint you were trying to call. Have a look at the API Documentation, found on every endpoints documentation page: https://sp-api-docs.saleweaver.com/redoc/vendorDirectFulfillmentOrdersV1.html

Your best bet will be the seller central support.

**Reply by robguttman** [reply]
Ah, sorry - I see the `headers` in the response object now.  Not sure how I missed that earlier.  Thanks for docs link - fyi, I'm actually using this one:

- https://sp-api-docs.saleweaver.com/endpoints/vendor_orders/
- https://sp-api-docs.saleweaver.com/redoc/vendorOrders.html

**Comment by robguttman** [comment]
Michael, so guess what - that same exact request that was failing before now succeeds:
```
VendorOrders().get_purchase_orders(changedAfter="2022-03-04T21:00:56.001000+00:00", nextToken=None)
```
So it does seem like there was some transient issue on Amazon's side.  But now I am well equipped to report issues to Amazon support thanks to the exposed `header` details.  Thanks again for your help!

**Comment by Nilesh-kodag** [comment]
I'm facing same problem for get_purchase_orders_status ()

    res = VendorOrders(credentials=CLIENT_CONFIG).get_purchase_orders_status(changedAfter="2022-03-05T15:15:48.001000+00:00", nextToken=None)


Receiving recent date data 
any idea how can i solve this
